src/base/AsyncJob.cc

```diff
@@ -12,14 +12,27 @@
 #include "base/AsyncCall.h"
 #include "base/AsyncJob.h"
 #include "base/AsyncJobCalls.h"
+#include "base/PackableStream.h"
 #include "base/TextException.h"
 #include "cbdata.h"
+#include "mem/PoolingAllocator.h"
 #include "MemBuf.h"
+#include "mgr/Registration.h"
+#include "Store.h"
 
 #include <ostream>
+#include <unordered_set>
 
 InstanceIdDefinitions(AsyncJob, "job");
 
+/// a set of all AsyncJob objects in existence
+static auto &
+AllJobs()
+{
+    static const auto jobs = new std::unordered_set<AsyncJob *, std::hash<AsyncJob *>, std::equal_to<AsyncJob *>, PoolingAllocator<AsyncJob *> >();
+    return *jobs;
```

Comment:
Some of the old Squid code does not allocate the object on heap, exposing us to `static` _destruction_ order [fiasco](https://isocpp.org/wiki/faq/ctors#construct-on-first-use-v2) risks. This implementation follows the recommended [practice](https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use) for use cases without important side effects in (never-called) destructors.

Most new code should follow this pattern for wrapping would-be-globals AFAICT.